### ISSUE TITLE
Ref always updated overriding local config value

### ISSUE THREAD

**Issue by ikerreyes** [RedHatInsights/bonfire#441]
I am struggling to make bonfire work with a local config file pointing to a remote repo as described in the [Readme](https://github.com/RedHatInsights/bonfire?tab=readme-ov-file#deploying-changes-changes-in-a-remote-git-branch).

To me is seems that the provided `ref` value gets always overriden by the ref value from the `--ref-env` or the fallback one at https://github.com/RedHatInsights/bonfire/blob/master/bonfire/qontract.py#L382

As a user, I would like to be able to use whatever reference env I set via `--ref-env` for most of my components except for the ones overridden by my local config.

When using the local config to point to a local host things work as expected, I guess because the `ref` is ignored.

**Comment by bsquizz** [comment]
The ref you provide for the component in the local config should override
the others. Some things to check ... Is it possible that the app or
component name used locally is not matching the name of the
application/resourceTemplate in app-interface?

Running `bonfire --debug <args...>` should also give us some more detailed
info about how the component's ref is being selected.

Can you give me an example config and the bonfire command you are running?

**Comment by ikerreyes** [comment]
I have tried the debug flag, but it has been tricky to find out anything related in the logs.

I am trying to deploy a component (`dvo-writer` which is part of `ccx-data-pipeline`) using the clowdapp template from this branch https://github.com/RedHatInsights/insights-results-aggregator/tree/poc-runtimes

The command I am using is `bonfire -d deploy ccx-data-pipeline --ref-env insights-production --frontends true -c local.yaml` being `local.yaml`

```
apps:
- name: ccx-data-pipeline
  components:
  - name: dvo-writer
    host: github
    repo: RedHatInsights/insights-results-aggregator
    path: /deploy/dvo-writer.yaml
    ref: poc-runtimes
```

Maybe I am doing a very obvious mistake, but the `local` source has worked fine.

**Comment by bsquizz** [comment]
I can reproduce what you're describing here and it *should* work the way you expect (as we document [here](https://github.com/RedHatInsights/bonfire?tab=readme-ov-file#deploying-changes-changes-in-a-remote-git-branch)) -- so, it seems there's a bug. Trying to sort it out.

**Comment by ikerreyes** [comment]
Thanks for checking @bsquizz Let me know if I can help anyhow

**Comment by bsquizz** [comment]
A fix is released in v6.0.1
